org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var https = require('https');
+var wh = require("opent2t-translator-helper-wink"); //link locally for now
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+var deviceId, accessToken;
+
+// module exports, implementing the schema
+module.exports = {
+
+    device: null,
+
+    initDevice: function(dev) {
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.access_token) !== 'undefined') {
+                    accessToken = props.access_token;
+                } else {
+                    console.log('props.access_token is undefined.');
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
```

Comment:
Yeah I wouldn't block this PR on that. We can always refactor out common code later. I agree eventually we'll probably want to have a helper module used by all translators. But it's probably a bit early for that now.

For validation, ideally we could just use some popular node argument-validation module... but surprisingly I couldn't find one.

For logging, I think we haven't figured out yet what should be the best practices for translators. The logging is really only beneficial for testing, not in a real application environment. For testing I think the logging could easily be much more verbose -- for example it could show the entire HTTP request/response headers and body, so when something goes wrong you can see all the details. But it should be disabled in an application environment.